packages/headless/src/controllers/controller/headless-controller.test.ts

```diff
@@ -10,9 +10,14 @@ describe('Controller', () => {
     return (engine.subscribe as jest.Mock).mock.calls.map((args) => args[0]);
   }
 
+  function updateControllerState(state: object) {
+    jest.spyOn(cmp, 'state', 'get').mockReturnValue(state);
+  }
+
   beforeEach(() => {
     engine = buildMockSearchAppEngine();
     cmp = buildController(engine);
+    updateControllerState({property: 'initial value'});
```

Comment:
More realistic initial mock state, these tests wouldn't pass now without the changes in the buildController method